### ISSUE TITLE
Postmortem rollout-ranker-quality-20260629c and add ranker quality diagnostics

### ISSUE THREAD

**Issue by NeonButrfly** [NeonButrfly/tichuml#113]
## Summary
The large live pipeline run `rollout-ranker-quality-20260629c` completed export, rollouts, training, and evaluation, but the challenger model failed badly against `server_heuristic` and the current training summary does not expose enough ranker-side baselines or decision-level diagnostics to explain why.

## Impact
- We can complete a long rollout-ranker loop mechanically and still end up with a model that is clearly not promotable.
- The current `rollout_ranker` training report surfaces `ndcg`, `pairwise_accuracy`, and `top_action_average_rollout_value`, but it does not compare those scores against simple ranker baselines.
- The finished evaluation also showed fallback contamination (`comparison_fallbacks=12`, `baseline_fallbacks=0`), which means the gate result is valid as a failure but not clean enough for fine-grained model-quality interpretation.
- Without postmortem tooling, we risk rerunning the same expensive loop without learning whether the main problem is weak labels, weak features, brittle runtime behavior, or a baseline that already encodes most of the available signal.

## Reproduction Steps
1. Inspect `/opt/tichuml/training-runs/rollout-ranker-quality-20260629c/ml/training-report.json`.
2. Inspect `/opt/tichuml/training-runs/rollout-ranker-quality-20260629c/evaluation-report.json`.
3. Observe that validation metrics exist for the ranker, but `baseline_metrics` is empty and there is no ranker-side model-vs-baseline comparison.
4. Observe that the mirrored evaluation gate failed decisively against `server_heuristic` and that challenger fallbacks were non-zero.

## Expected vs Actual
Expected:
- failed ranker runs should produce enough diagnostics to tell whether the model beat simple internal baselines, which decision groups it gets wrong, and whether evaluation contamination weakens the conclusion
- postmortem artifacts should clearly distinguish model-quality failure from infrastructure/runtime contamination

Actual:
- the run proves the candidate is not promotable, but the current report leaves too much ambiguity about whether the training signal itself is weak or whether the model merely failed to beat trivial ranker heuristics

## Root Cause
Known so far:
- the current `rollout_ranker` report path lacks simple baselines analogous to the regression baselines added for `observed_outcome_regression`
- the current postmortem surface does not emit a concrete bad-decision slice or summary report for failed ranker runs
- evaluation allowed challenger fallbacks, so the failure signal is directionally real but partially contaminated

Unknown but to diagnose:
- whether the primary weakness is target construction, missing features, feature over-reliance on broad state descriptors like `action_rank`, or normal weakness of the current rollout label regime

## Proposed Fix
- add ranker baseline metrics to `ml/train_lightgbm.py`
- add a ranker postmortem/diagnostic script under `scripts/` that inspects a finished run and summarizes validation, baselines, top mistakes, and evaluation contamination
- update docs and training summary output so failed ranker runs are easier to interpret before re-launching another large loop
- document the concrete findings from `rollout-ranker-quality-20260629c`

## Validation Plan
- add focused failing tests for ranker baseline reporting and postmortem outputs
- run the relevant integration tests locally
- run the new diagnostic against `rollout-ranker-quality-20260629c` or an equivalent local fixture
- verify docs and issue updates match the implemented behavior

## Prompt Capture Seed
Prompt request: do a concrete postmortem on the failed rollout-ranker run, then do the remediation work in order and iterate until complete instead of launching another blind large training loop.

**Comment by NeonButrfly** [comment]
Implemented the first concrete postmortem/remediation pass for the failed `rollout-ranker-quality-20260629c` run.

What landed locally:
- `ml/train_lightgbm.py`
  - added ranker baseline metrics for `rollout_ranker`
  - baselines now include:
    - `action_rank_descending`
    - `grouped_by_action_type_mean_target`
    - `grouped_by_seat_action_type_mean_target`
  - added ranker-side `model_vs_baseline` comparison logic instead of leaving `baseline_metrics` empty
  - training markdown now prints all available baseline validation metrics instead of assuming RMSE/MAE-only regression output
- `scripts/rollout_ranker_postmortem.py`
  - new finished-run diagnostic for rollout-ranker runs
  - loads run metadata + saved model + source dataset
  - recomputes validation metrics on the stored split strategy
  - summarizes internal baselines, evaluation fallback contamination, and highest-regret validation decisions
  - writes:
    - `diagnostics/rollout_ranker_postmortem.json`
    - `diagnostics/rollout_ranker_postmortem.md`
- docs:
  - `docs/ml-strategy-improvement.md` now documents the new ranker baselines and the rollout-ranker postmortem script
- tests:
  - expanded `tests/integration/ml-export-train-regression.test.ts` to assert ranker baselines/model-vs-baseline output
  - added `tests/integration/rollout-ranker-postmortem.test.ts`
  - updated `tests/integration/outcome-reward-diagnostics.test.ts` to parse the final JSON payload line from trace-emitting scripts

Validation performed locally:
- `PYTHON_EXECUTABLE=/Users/admin/.cache/codex-runtimes/codex-primary-runtime/dependencies/python/bin/python3 node node_modules/vitest/vitest.mjs run tests/integration/lightgbm-training-summary.test.ts tests/integration/outcome-reward-diagnostics.test.ts tests/integration/ml-export-train-regression.test.ts tests/integration/rollout-ranker-postmortem.test.ts`
- result: `4` files passed, `14` tests passed

Important interpretation change from this pass:
- future failed `rollout_ranker` runs will no longer stop at “validation metrics exist but baseline_metrics is empty”
- we can now see whether the model is actually beating trivial internal heuristics before spending more time on another large rollout/training loop

Next recommended follow-up after this issue update:
1. run `scripts/rollout_ranker_postmortem.py` against the real `rollout-ranker-quality-20260629c` artifacts on the Linux host
2. inspect whether the model loses to `action_rank_descending` / grouped means, or whether the larger problem is evaluation fallback contamination
3. only then decide whether the next loop should be feature work, target work, or runtime/eval hardening

**Comment by NeonButrfly** [comment]
Implemented the next training-signal hardening step after the host-side diagnosis.

What landed locally:
- `ml/train_lightgbm.py`
  - added `--min-rollout-samples`
  - added `--min-rollout-stddev`
  - rollout objectives can now filter weak decision groups before training based on:
    - minimum rollout sample count per decision
    - minimum rollout stddev per decision
  - training metadata/report output now records:
    - `min_rollout_samples`
    - `min_rollout_stddev`
    - `filtered_out_low_sample_decision_count`
    - `filtered_out_low_stddev_decision_count`
- `scripts/ml-live-bootstrap.ts`
  - live bootstrap plan now accepts and forwards the same rollout-quality gates into the `ml:train` step
- tests:
  - added trainer coverage for low-sample / low-stddev rollout filtering
  - updated live-bootstrap coverage to assert the new train-step args are propagated when configured

Why this step:
- the real Linux run showed a huge raw export (`1.19M` candidate rows) collapsing to a much smaller rollout-labeled subset (`18,876` rows across `1,614` decisions)
- the next safe improvement is to tighten label quality before another expensive loop, not just rerun larger
- this gives us a controllable way to exclude obviously weak rollout-label decisions from future ranker training

Validation performed locally:
- `PYTHON_EXECUTABLE=/Users/admin/.cache/codex-runtimes/codex-primary-runtime/dependencies/python/bin/python3 node node_modules/vitest/vitest.mjs run tests/integration/lightgbm-training-summary.test.ts tests/integration/outcome-reward-diagnostics.test.ts tests/integration/ml-export-train-regression.test.ts tests/integration/rollout-ranker-postmortem.test.ts tests/integration/ml-live-bootstrap.test.ts`
- result: `5` files passed, `22` tests passed

Recommended next operational step after this lands:
1. sync Linux host `/opt/tichuml`
2. rerun the bounded live bootstrap / rollout-ranker loop with nonzero rollout-quality gates, for example a first pass using:
   - `--min-rollout-samples 2`
   - `--min-rollout-stddev 10`
3. evaluate whether that cleaned subset improves internal baselines and head-to-head results before scaling the run up further
